jakarta-eclipse/src/org/jakarta/codeAction/MicroProfileJavaCodeActionParams.java

```diff
@@ -0,0 +1,33 @@
+package org.jakarta.codeAction;
+
+import org.eclipse.lsp4j.CodeActionContext;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+
+public class MicroProfileJavaCodeActionParams extends CodeActionParams {
+
+	private boolean resourceOperationSupported;
+
+	public MicroProfileJavaCodeActionParams() {
+		super();
+	}
+
+	public MicroProfileJavaCodeActionParams(final TextDocumentIdentifier textDocument, final Range range,
+			final CodeActionContext context) {
+		super(textDocument, range, context);
+	}
+
+	public String getUri() {
+		return getTextDocument().getUri();
+	}
+
+	public boolean isResourceOperationSupported() {
+		return resourceOperationSupported;
+	}
+
+	public void setResourceOperationSupported(boolean resourceOperationSupported) {
+		this.resourceOperationSupported = resourceOperationSupported;
+	}
+
+}
```

Comment:
I noticed that we have diagnosticParams in use, but it seems to have fewer properties compared to this params class. Should I keep this param class for code action specifically? @kathrynkodama @rzgry

Reply:
This looks to be a separate class for CodeActionsParams, it should be separate from `diagnosticParams`. Let me know if I am misunderstanding your question.

Reply:
yea it is the answer I need. Also just to double check, do I need to give the file a different name. I am having this file under the codeaction package I created. How about I renamed the file to JakartaJavaCodeActionParams.java?

Reply:
Sure sounds good